### DISCUSSION TITLE
Welcome to Slurm-Mail Discussions!

### DISCUSSION THREAD

**Discussion by neilmunday** [neilmunday/slurm-mail#14] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mtblondie** [comment]
I have a small thing I'd like to resolve - if you can help. 
Im using the plugin, everything is installed and configured and working as expected. 
my email is firstname.lastname. When my email comes in, it says Dear LASTNAME, and thats all. 
How and where can I change that configuration?

**Reply by neilmunday** [reply]
Hi,

The `$USER` value that is used in the e-mail templates comes from the [GECOS](https://en.wikipedia.org/wiki/Gecos_field) field of `/etc/passwd`.

Slurm-Mail used the `pwd` module to get the [GECOS](https://en.wikipedia.org/wiki/Gecos_field) field. The [GECOS](https://en.wikipedia.org/wiki/Gecos_field) is typically a comma separated list of strings where the first item is the user's full name.

Slurm-Mail therefore splits the [GECOS](https://en.wikipedia.org/wiki/Gecos_field) by comma and extracts the first item in the returned list to use as the user's full name.

Are you able to share how your GECOS field is structured?

For example, you could run this code:

```python
#!/usr/bin/env python3

if __name__ === "__main__":
    print(pwd.getpwnam("YOUR_USERNAME_HERE").pw_gecos)

**Reply by mtblondie** [reply]
#!/usr/bin/env python3

import pwd

name = "byrnka"

if __name__ == "__main__":
    print(pwd.getpwnam(name).pw_gecos)


# ./username_test.py 
Byrne, Kelly Anne

**Reply by neilmunday** [reply]
Thanks for the info. I'll see if I can come up with a configuration option that would allow admins of Slurm-Mail to specify how the GECOS field should be treated.

**Reply by neilmunday** [reply]
Tracking this feature request in: https://github.com/neilmunday/slurm-mail/issues/133

**Reply by neilmunday** [reply]
@mtblondie version 4.19 has now been released.

Once installed, please add to `/etc/slurm-mail/slurm-mail.conf` in the `[slurm-send-mail]` section:

```
gecosNameField = 1
```

This should then change the value used in the e-mails you receive from surnames to first names.

**Reply by neilmunday** [reply]
@mtblondie were you able to try out the new version?

**Reply by mtblondie** [reply]
Im sorry _ I just got back to this.   Im doing it right now, I'll let you know shortly. =D

**Reply by mtblondie** [reply]
Yes!!! Its working.  Thank you so much

**Reply by neilmunday** [reply]
Glad to hear the fix worked for you :-)

**Comment by aav18141** [comment]
Hi! 
Is it possible to add GPU number to the templates?
In scontrol output it appers in *TRES fields:

   NumNodes=1 NumCPUs=6 NumTasks=1 CPUs/Task=1 ReqB:S:C:T=0:0:*:*
   ReqTRES=cpu=5,mem=90000M,node=1,billing=5,gres/**gpu=1** 
   AllocTRES=cpu=6,mem=90000M,node=1,billing=6,gres/**gpu=1**

**Reply by neilmunday** [reply]
Hi,

Thanks for the suggestion. I have created https://github.com/neilmunday/slurm-mail/issues/142

**Reply by neilmunday** [reply]
Please see discussion in https://github.com/neilmunday/slurm-mail/issues/142, I would be interested to know your thoughts.

**Reply by neilmunday** [reply]
@aav18141 version 4.22 released which includes the new TRES feature
